### PR TITLE
feat: added news support to the status notice

### PR DESCRIPTION
**Pull request by ice-hades**

### CODE REVIEW

**Review by ice-ares — DISMISSED** [review]

**Review by ice-orion — DISMISSED** [review]

**Review by ice-orion — DISMISSED** [review]

**Review by ice-orion — APPROVED** [review]

File: src/store/modules/News/reducer/index.ts [hunk 1]
```diff
@@ -82,8 +82,9 @@ export function newsReducer(state = INITIAL_STATE, action: Actions): State {
       case NewsActions.NEWS_ARTICLE_MARK_VIEWED(null).SUCCESS.type:
         {
           const {newsId} = action.payload;
-
-          draft.items[newsId].viewed = true;
+          if (draft.items[newsId]) {
```
Comment (ice-atlas) [inline src/store/modules/News/reducer/index.ts:85]: why we need this? You can not open/mark as viewed article that does not exist

File: src/store/modules/StatusNotice/sagas/checkStatusNoticeSaga.ts [hunk 2]
```diff
@@ -24,22 +24,33 @@ export function* checkStatusNoticeSaga() {
...
-      yield put(
-        StatusNoticeActions.SET_STATUS_NOTICE_DATA.STATE.create({
-          data: statusNoticeData,
-        }),
-      );
     }
+  }
+  if (noticeData?.data) {
+    statusNoticeData.data =
+      noticeData.data[user?.language ?? ''] ?? noticeData.data.en;
```
Comment (ice-atlas) [inline src/store/modules/StatusNotice/sagas/checkStatusNoticeSaga.ts:46]: `noticeData.data[user?.language ?? ''] ?? noticeData.data.en;` -> `noticeData.data[user?.language ?? 'en'];` is not the same?

File: src/store/modules/StatusNotice/sagas/checkStatusNoticeSaga.ts [hunk 3]
```diff
@@ -24,22 +24,33 @@ export function* checkStatusNoticeSaga() {
...
-        content: localisedData.content,
+  const statusNoticeData: StatusNoticeData = {
+    link: noticeData?.link,
+    gradientColors: noticeData?.gradientColors,
+    icon: noticeData?.icon,
+  };
+  if (noticeData?.newsData) {
+    const localisedNewsData =
+      noticeData.newsData.localisedData[user?.language ?? ''] ??
+      noticeData.newsData.localisedData.en;
```
Comment (ice-atlas) [inline src/store/modules/StatusNotice/sagas/checkStatusNoticeSaga.ts:35]: same here `noticeData.newsData.localisedData[user?.language ?? 'en']`

File: src/store/modules/StatusNotice/sagas/checkStatusNoticeSaga.ts (hunk 2 above)
Comment (ice-hades) [reply src/store/modules/StatusNotice/sagas/checkStatusNoticeSaga.ts:46]: not the same. you can have a language which is not null but is not present in the notice data

File: src/store/modules/StatusNotice/sagas/checkStatusNoticeSaga.ts (hunk 3 above)
Comment (ice-hades) [reply src/store/modules/StatusNotice/sagas/checkStatusNoticeSaga.ts:35]: same answer

File: src/store/modules/News/reducer/index.ts (hunk 1 above)
Comment (ice-hades) [reply src/store/modules/News/reducer/index.ts:85]: we can get an article to track from the push

Comment (ice-hades) [reply src/store/modules/News/reducer/index.ts:85]: and from the status notice in this case

Comment (ice-hades) [reply src/store/modules/News/reducer/index.ts:85]: point is that article data may not be in the redux data

Comment (ice-atlas) [reply src/store/modules/News/reducer/index.ts:85]: Is there only `articleId` or whole data that can be added to state as well?

Comment (ice-hades) [reply src/store/modules/News/reducer/index.ts:85]: not the whole data

**Review by ice-atlas — APPROVED** [review]

**Pull request merged**